aten/src/ATen/Device.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <ATen/Error.h>
+#include <ATen/optional.h>
+
+#include <cstddef>
+#include <iosfwd>
+#include <string>
+
+namespace at {
+enum class Backend;
+struct Tensor;
+} // namespace at
+
+namespace at {
+
+/// Represents a a compute device on which a tensor is located. A device is
+/// uniquely identified by a type, which specifies the type of machine it is
+/// (e.g. CPU or CUDA GPU), and a device index or ordinal, which identifies the
+/// specific compute device when there is more than one of a certain type. The
+/// device index is optional, and in its defaulted state represents (abstractly)
+/// "the current device". Further, there are two constraints on the value of the
+/// device index, if one is explicitly stored:
+/// 1. The device index is never negative.
+/// 2. When the device type is CPU, the device index must be zero.
+struct Device {
```

Comment:
So from convergence standpoint - this is pretty much DeviceOption from Caffe2. No action needed, just pointing out